src/main/java/org/pih/petl/job/type/SqlServerImportJob.java

```diff
@@ -137,14 +138,19 @@ public void execute(final ExecutionContext context) throws Exception {
                         log.debug("Executing: " + sqlStatement);
                         StopWatch sw = new StopWatch();
                         sw.start();
-                        statement = sourceConnection.createStatement();
-                        statement.execute(sqlStatement);
-                        log.debug("Statement executed");
-                        if (!sqlIterator.hasNext()) {
+                        if (sqlIterator.hasNext()) {
+                            statement = sourceConnection.createStatement();
+                            statement.execute(sqlStatement);
+                            log.debug("Statement executed");
+                        }
+                        else {
                             log.debug("This is the last statement, treat it as the extraction query");
+                            statement = sourceConnection.prepareStatement(
+                                    sqlStatement, ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY
+                            );
                             ResultSet resultSet = null;
                             try {
-                                resultSet = statement.getResultSet();
+                                resultSet = ((PreparedStatement)statement).executeQuery();
```

Comment:
The "resultSet.last()" mechanism used below no longer works by default with the MySQL 8 connector.  You need to instead use a prepared statement and instruct it to allow resultset scrolling.  This is the nature of the above changes.